agent_plugin_builder/build_plugin.py

```diff
@@ -0,0 +1,269 @@
+import json
+import logging
+import shutil
+import tarfile
+from importlib import import_module
+from pathlib import Path
+from typing import Callable
+
+import yaml
+from monkeytypes import AgentPluginManifest
+
+from .build_options import PlatformDependencyPackagingMethod, parse_agent_plugin_build_options
+from .vendor_dirs import (
+    check_if_common_vendor_dir_possible,
+    generate_common_vendor_dir,
+    generate_requirements_file,
+    generate_vendor_dirs,
+)
+
+logger = logging.getLogger(__name__)
+
+
+BUILD = "build"
+DIST = "dist"
+CONFIG_SCHEMA = "config-schema.json"
+SOURCE = "source"
+MANIFEST = "manifest"
+
+
+def build_agent_plugin(
+    plugin_path: Path,
+    build_dir_path: Path,
+    dist_dir_path: Path,
+    on_build_dir_created: Callable[[Path], None] | None = None,
```

Comment:
This complicates stuff. We don't need a callable for the log file. The `build_agent_plugin` will eventually be exposed and it needs to be simple.

Reply:
It is an _optional_ argument -- one need not provide it